java/src/test/java/org/rocksdb/RocksDBTest.java

```diff
@@ -209,6 +210,197 @@ public void merge() throws RocksDBException {
           "xxxx".getBytes());
     }
   }
+ private class M extends AbstractAssociativeMergeOperator{
```

Comment:
Can you move this and the associated tests to its own test file?

Reply:
Ok tests works but there is a problem to solve for having the best performance. 
Threads lanched by rocksdb are not terminated in synchronous way with "delete db"(pratically rocks threads are terminated after so it happens jvm is closed before rocksdb are closed creating a memory leak. For evoiding it for now i create context as in old code. This problem appear just with rocksdb threads not with java threads. ). 
For solving the problem i might add a simple method in posix-env joining all threads inside thread pool.